src/components/backend/vectorDB.js

```diff
@@ -21,27 +20,66 @@ class VectorDB {
         const vectorObjs = await this.indexedDB.object("embedding").getAll();
         const vectors = vectorObjs.map(v => v.vector);
         const documents = vectorObjs.map(v => {
-            return { pageContent: '', metadata: { id: v.id } }
+            return { pageContent: '', metadata: {
+                id: v.id,
+                offset: v.offset } }
         })
         this.db.addVectors(vectors, documents);
     }
 
+    async splitText(text, chunkSize, chunkOverlap, 
```

Comment:
curious about what does `chunkSize` and `chunkOverlap` do? I played around the separators and can probably simplify / optimize performance with built in `string.split` and match with regular expression. Something like ```str.split(/(##|\||>|-|\n+|\.|\?|\!)/).filter(r => r !== "");``` This generate an array with separated text as well as the separators, so we can also retroactively back-fill the `offset` array if we need to store it.